spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe OmniauthCallbacksController, :type => :controller do
+  before(:each) do
+    request.env["devise.mapping"] = Devise.mappings[:user]
+  end
+
+  describe "#facebook" do
+    it "should successfully authenticate using Facebook authentication hash and display a success message in flash, if the user persists" do
+      request.env["omniauth.auth"] = FactoryGirl.create(:auth_hash, :facebook)
+      get :facebook
+      user = User.find_by(email: 'testuser@facebook.com')
+      expect(user).not_to eq(nil)
+      expect(flash[:notice]).to eq("Successfully authenticated from Facebook account.")
```

Comment:
I think we should also make sure that the user actually gets signed in during the successful authentication tests. This is pretty easy to do via `subject` (http://www.relishapp.com/rspec/rspec-core/v/3-4/docs/subject/implicitly-defined-subject) and the `current_user` helper provided by Devise.